packages/client/src/DataUnion.ts

```diff
@@ -487,18 +521,27 @@ export class DataUnion {
         }
 
         const adminFeeBN = BigNumber.from((newFeeFraction * 1e18).toFixed()) // last 2...3 decimals are going to be gibberish
-        const ethersOverrides = this.client.getOverrides()
-        let tx
-        try {
-            tx = await this.contract.setAdminFee(adminFeeBN, ethersOverrides)
-        } catch(error) {
-            if (error.message.includes('error_onlyOwner')) {
-                const myAddress = await this.contract.signer.getAddress()
-                throw new Error(`Setting admin fee for data union ${this.contract.address} failed: ${myAddress} is not the DataUnion admin!`)
-            }
-            throw error
-        }
-        return waitOrRetryTx(tx)
+        return this.sendAdminTx(this.contract.setAdminFee, adminFeeBN)
+    }
+
+    /**
+     * Stores a Javascript object in JSON format in the data union contract, can be retrieved with `getMetadata`
+     * @param metadata object to be stored in the data union contract
+     */
+    async setMetadata(metadata: object): Promise<ContractReceipt> {
+        return this.sendAdminTx(this.contract.setMetadata, JSON.stringify(metadata))
+    }
+
+    async setMetadataString(metadata: string): Promise<ContractReceipt> {
```

Comment:
Probably not necessary / good to expose a method to the end-user that enables them to break things. Better to enforce writing JSON to the metadata field.

Reply:
ok so just remove the those "raw string" methods?

Reply:
I was thinking it could be a URL or something, but of course it could just as well be wrapped in JSON  🤷

Reply:
> ok so just remove the those "raw string" methods?

Yeah I would say so